src/main/java/org/terasology/weatherManager/clouds/CloudProvider.java

```diff
@@ -56,7 +57,7 @@ public void process(GeneratingRegion region) {
 
         float anim = cloudManager.getAnimFrame();
 
-        Rect2i reg = facet.getWorldRegion();
+        BlockAreac reg = facet.getRelativeArea();
```

Comment:
I think the following loops can be simplified by just iterating over the region:

```suggestion
for (Vector2ic pos: facet.getRelativeArea()) {
	if (isClouded(pos.x(), pos.y(), anim)) {
    	facet.setWorld(pos, true);
    }
}
```